src/components/Launch/LaunchWorkflowForm/inputHelpers/utils.ts

```diff
@@ -25,3 +25,36 @@ export function collectionChildToString(type: InputType, value: any) {
     }
     return type === InputType.Integer ? `${value}` : JSON.stringify(value);
 }
+
+/** Determines if a given input type, including all levels of nested types, is
+ * supported for use in the Launch form.
+ */
+export function typeIsSupported(typeDefinition: InputTypeDefinition): boolean {
+    const { type, subtype } = typeDefinition;
+    switch (type) {
+        case InputType.Map:
+        case InputType.None:
+        case InputType.Schema:
+        case InputType.Unknown:
+            return false;
+        case InputType.Boolean:
+        case InputType.Datetime:
+        case InputType.Duration:
+        case InputType.Float:
+        case InputType.Integer:
+        case InputType.String:
+            return true;
+        case InputType.Collection: {
+            if (!subtype) {
+                console.error(
+                    'Unexpected missing subtype for collection input',
+                    typeDefinition
+                );
+                return false;
+            }
+            return typeIsSupported(subtype);
+        }
+        default:
```

Comment:
Depending on requirements here, we could also consider asserting never in the default block to make sure this switch covers all `InputType`s at compile time

Reply:
Oh I like that. Thanks!